src/vsr/replica.zig

```diff
@@ -1746,13 +1820,39 @@ pub fn ReplicaType(
             self.send_message_to_replica(replica, prepare.message);
         }
 
-        fn on_commit_timeout(self: *Self) void {
-            self.commit_timeout.reset();
+        fn on_primary_abdicate_timeout(self: *Self) void {
+            assert(self.status == .normal);
+            assert(self.primary());
+            assert(self.primary_pipeline_pending() != null);
+            self.primary_abdicate_timeout.reset();
```

Comment:
Feels a bit weird that we do a `.reset()` here rather than a `.stop()`, as any further ticking is useless. At the same time, this makes it much easier to formulate when the timeout should be ticking. I think I like the current approach more, even if it is wasteful in some sense.